Add VH powheg cards for 2017

alpakpinar opened the pull request.

agrohsje commented:
Dear @alpakpinar . Please upload only one example and a script that adjusts the necessary options in the template.

alpakpinar commented:
Dear @agrohsje , okay I will then add only one additional card for each process and slightly modify the script so that it adjusts the template accordingly, thanks.

alpakpinar commented:
Dear @agrohsje , I now updated the pull request regarding the VH cards. For each process (WminusH, WplusH, ZH and ggZH), I only have one example card, for mH=125. I also updated the script that generates the powheg cards:

- The script now can read the mass points and the decay mode for the V-boson from the command line (previously there were separate scripts for separate decay modes)
- It adjusts the required parameters in the Powheg input card accordingly with the given Higgs mass

Do you think the cards & the script are okay to merge with these changes? Let me know if you want me to make additional changes, thanks.

alpakpinar commented:
Dear @agrohsje , I now also added similar code to generate 2016 powheg input cards for ggHZ. I added one script for card generation, one card template and one example card with mH=125. Do you think I should do additional updates, or can we merge this PR if this looks okay? Thank you!

agrohsje commented:
Please remove: 
rwl_group_events 2000
lhapdf6maxsets 50
rwl_file 'pwg-rwl.dat'
rwl_format_rwgt 1

They should be added by the scripts automatically.

alpakpinar commented:
Hi @agrohsje , now these four lines are removed from the end of the powheg input cards. For each VH process, the relevant makecards.py script now appends these four lines at the end of the generated input card automatically.

agrohsje commented on bin/Powheg/production/2017/13TeV/Higgs/HZJ_HanythingJ_NNPDF31_13TeV/HZJ_HanythingJ_NNPDF31_13TeV_M125_Vhadronic.input:
```diff
@@ -74,3 +103,7 @@ doublefsr 1
 
 nohad   1
 
+rwl_group_events 2000
+lhapdf6maxsets 50
+rwl_file 'pwg-rwl.dat'
+rwl_format_rwgt 1
```
seems it is still there.

alpakpinar commented:
Okay, now those lines should be removed from the example input cards as well, the scripts won't append those lines to the generated cards.

agrohsje commented:
Dear @alpakpinar ,
I still have a few questions and suggestions. 
1.) Why do you disable withnegweights ? I think it is good to keep and consider the neg. weights as you know how large the contribution is.  
2.) I would consistently, i.e. in all cards, remove unnecessary lines like:
storeinfo_rwgt 0 
pdfreweight 0 
see e.g. here https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/run_pwg.py#L938
3.) Moreover, I suggest to remove the new lines that you added but start with ! or # . There is no reason to add additional lines that are anyway uncommented. 
As I will be offline for 2 weeks. Maybe @covarell and @qliphy can follow-up on your next round of commits. 
Cheers, Alexander.

alpakpinar commented:
Hi @agrohsje, @covarell , @qliphy. Thanks for the suggestions, I now added new commits to address these issues.

About the negative weights, I think removing them was a mistake, I now re-enabled the storing of negative weights in all cards.

I also removed some unnecessary lines like storeinfo_rwgt 0 and pdfreweight 0, as @agrohsje suggested. I also removed the lines that are commented out. Thanks!

alpakpinar commented:
Sorry for the multiple messages in the thread, I wanted to ask a quick question regarding the ttH cards, as I'm also currently working on those with different Higgs mass points. Higgs mass points range from 110 GeV to 1 TeV.   

I wanted to ask about the parameter in powheg input cards for ttH, which is in [1]. Considering the Higgs mass point variations, how can I vary this parameter? If I'm understanding correctly, for small variations of Higgs mass, it is fine to keep it as a constant, but I'm not sure how to handle the situation when the mass variations are large.

Thanks so much for the help!
Alp

[1]: https://github.com/alpakpinar/genproductions/blob/2020-07-14_vh_powheg_cards/bin/Powheg/production/2017/13TeV/Higgs/ttH_inclusive_hdamp_NNPDF31_13TeV_MHVar/ttH_inclusive_hdamp_NNPDF31_13TeV_M130.input#L5

alpakpinar commented:
Hi @agrohsje, just wanted to ask if the PR looks okay after the latest changes. Please let me know if you would like further updates and I can work on them. Thanks!

Alp

agrohsje commented:
+1 @qliphy
